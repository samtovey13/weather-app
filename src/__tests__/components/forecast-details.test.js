import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from '../../components/forecast-details';
import ForecastSummaries from "../../components/forecast-summaries";

describe("ForecastDetails", () => {

  const forecast = {
    date: 1525219200000,
    temperature: {
      max: 999,
      min: 0,
    },
    humidity: 80,
    wind: {
      speed: 20,
      direction: "se"
    }
  };

  it("renders correctly", () => {

    const { asFragment } = render(<ForecastDetails 
      date={forecast.date}
      temperature={forecast.temperature}
      humidity={forecast.humidity}
      wind={forecast.wind}
    />);
    
    expect(asFragment()).toMatchSnapshot();
  })

  it("renders the correct date, icon, description and temperature props", () => {
    
    const { getByText } = render(
      <ForecastDetails 
      date={forecast.date}
      temperature={forecast.temperature}
      humidity={forecast.humidity}
      wind={forecast.wind} 
      />
    );

    expect(getByText("Wed 2nd May")).toHaveClass("date");
    expect(getByText("Min temp: 0°c")).toHaveClass("min-temperature");
    expect(getByText("Max temp: 999°c")).toHaveClass("max-temperature");
    expect(getByText("Humidity: 80%")).toHaveClass("humidity");
    expect(getByText("Wind speed: 20mph")).toHaveClass("wind-speed");
    expect(getByText("Wind direction: SE")).toHaveClass("wind-direction");

  });

})