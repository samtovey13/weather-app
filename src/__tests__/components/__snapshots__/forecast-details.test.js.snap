// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <span
      class="date"
      data-testid="date-id"
    >
      Wed 2nd May
    </span>
    <span
      class="min-temperature"
      data-testid="min-temperature-id"
    >
      Min temp: 0°c
    </span>
    <span
      class="max-temperature"
      data-testid="max-temperature-id"
    >
      Max temp: 999°c
    </span>
    <span
      class="humidity"
      data-testid="humidity-id"
    >
      Humidity: 80%
    </span>
    <span
      class="wind-speed"
      data-testid="wind-speed-id"
    >
      Wind speed: 20mph
    </span>
    <span
      class="wind-direction"
      data-testid="wind-direction-id"
    >
      Wind direction: SE
    </span>
  </div>
</DocumentFragment>
`;
